src/Core/src/PropertyMapper.cs

```diff
@@ -93,35 +126,55 @@ public IPropertyMapper[]? Chained
 			}
 		}
 
-		private HashSet<string> PopulateKeys()
-		{
-			var keys = new HashSet<string>(StringComparer.Ordinal);
-			foreach (var key in GetKeys())
-			{
-				keys.Add(key);
-			}
-			return keys;
-		}
-
+		// TODO: Make private in .NET10 with a new name: ClearMergedMappers
 		protected virtual void ClearKeyCache()
 		{
-			_updateKeys = null;
+			_mergedMappers = null;
+			_mergedKeys = null;
 		}
 
-		public virtual IReadOnlyCollection<string> UpdateKeys => _updateKeys ??= PopulateKeys();
+		// TODO: Remove in .NET10
+		public virtual IReadOnlyCollection<string> UpdateKeys => MergedKeys;
 
 		public virtual IEnumerable<string> GetKeys()
 		{
-			foreach (var key in _mapper.Keys)
-				yield return key;
-
+			// We want to retain the initial order of the keys to avoid race conditions
+			// when a property mapping is overridden by a new instance of property mapper.
+			// As an example, the container view mapper should always run first.
+			// Siblings mapper should not have keys intersection.
 			if (Chained is not null)
 			{
-				foreach (var chain in Chained)
-					foreach (var key in chain.GetKeys())
+				for (int i = Chained.Length - 1; i >= 0; i--)
```

Comment:
I feel a bit dumb here, but why do we reverse the lookup?

Reply:
https://github.com/dotnet/maui/blob/69a166b7eebc480630c8d63870e4143ffdd71a41/src/Core/src/Handlers/Button/ButtonHandler.cs#L36-L37

We usually give execution priority to the leftmost, but for keys, we have to look at the rightmost to see what has been defined first.

Like the `MapContainer` to initialize the container view should be the first one, and it's defined in the`ViewHandler.Mapper`.